### PR TITLE
Drawing

### PR DESCRIPTION
**Pull request by stdavis**
Next up is gathering and submitting data to the new feature service.

### CODE REVIEW

**Comment by steveoh** [comment]
After clicking add feature, draw, finish, add action, cancel - I am taken to an empty Feature tab and I am unable to click add feature to start the wizard again.

**Comment by steveoh** [comment]
Another thing to keep in mind would be the authentication piece. Maybe start to flesh out when these things are created based on the values in the page.

**Comment by steveoh** [comment]
I noticed you are able to create duplicate action/treaments

image

**Comment by steveoh** [comment]
Cut tool is not necessary for point features.

image

**Comment by stdavis** [comment]
> After clicking add feature, draw, finish, add action, cancel - I am taken to an empty Feature tab and I am unable to click add feature to start the wizard again.

Addressed with: 48d43b2

**Comment by stdavis** [comment]
> I noticed you are able to create duplicate action/treaments

Addressed with: 7500e62

**Comment by steveoh** [comment]
we can use these to test out the permissions

image

**Comment by stdavis** [comment]
> Cut tool is not necessary for point features.

Addressed with fddf3e9

**Comment by stdavis** [comment]
> we can use these to test out the permissions
> 
> 
> 
> maybe a separate issue but something to keep in mind.

Can they put an id on that form to make it cleaner for us to get a reference to it?

If you are OK with it I'd like to get this PR merged so that we can start new with permissions and submitting the data to the service.

I've done a fresh rebase against master.

File: src/app/project/FeatureDetails.js [hunk 1]
```diff
@@ -25,6 +26,7 @@ define([
     _TemplatedMixin,
     _WidgetBase,
 
+    domClass,
     domConstruct,
     query,
```
Comment (steveoh) [inline src/app/project/FeatureDetails.js]: query looks to be unused

**Comment by steveoh** [comment]
What are your thoughts on if you select a point feature type, draw, and switch to another point feature type - do you think the points should be cleared?

File: src/app/project/Action.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        treatment: null,
+
+        // retreatment: String (true | false) (optional)
+        //      string value of the checked state of the checkbox
+        retreatment: null,
+
+        // action: String
+        action: null,
+
+        // comments: String (optional)
```
Comment (steveoh) [inline src/app/project/Action.js]: we are going to have to add `herbicide`

**Comment by steveoh** [comment]
If a user selects `Herbicide Action` then populate a select with the values from `lu_herbicide` only for `terrestrial` and `aquatic` treatments.

**Comment by stdavis** [comment]
> What are your thoughts on if you select a point feature type, draw, and switch to another point feature type - do you think the points should be cleared?

Yes, because all of the other controls are cleared also. I've also talked this UX over with Aaron.

File: src/app/project/FeatureDetails.js (hunk 1 above)
Comment (stdavis) [inline src/app/project/FeatureDetails.js]: Fixed in 914e606

File: src/app/project/NewFeatureWizard.js [hunk 3]
```diff
@@ -275,6 +399,100 @@ define([
...
+            });
+        },
+        onCancel: function () {
+            // summary:
+            //      destroy this widget and all associated data
+            console.log('app.project.NewFeatureWizard:onCancel', arguments);
+
+            this.graphicsLayer.clear();
+
+            this.emit("hide");
```
Comment (steveoh) [inline src/app/project/NewFeatureWizard.js]: single quotes!

File: src/app/project/NewFeatureWizard.js [hunk 4]
```diff
@@ -275,6 +399,100 @@ define([
...
+                params.comments = this.commentsTxt.value;
+            }
+            if (visible(this.retreatment)) {
+                params.retreatment = this.retreatmentChBx.checked;
+            }
+
+            var existing = this.actions.some(function (a) {
+                var match;
+                for (var p in params) {
+                    if (params.hasOwnProperty(p)) {
```
Comment (steveoh) [inline src/app/project/NewFeatureWizard.js]: use Object.keys here

File: src/jsonconfig/config.json [hunk 5]
```diff
@@ -1,355 +1,355 @@
...
+            "Draft", 
             1
-        ],
+        ], 
         [
-            "Proposed",
+            "Proposed", 
             2
-        ],
+        ],
```
Comment (steveoh) [inline src/jsonconfig/config.json]: a lot of trailing white spaces in here? What is this file for?

File: src/app/project/FeatureDetails.js [hunk 6]
```diff
@@ -103,14 +107,24 @@ define([
...
-            this.featureTabLink.click();
-
-            var wizard = new NewFeatureWizard({}, this.featureTabContents);
-            wizard.startup();
+            // show new feature tab
+            domClass.remove(this.newFeatureTab, 'hidden');
+            domClass.remove(this.newFeatureTabContainer, 'hidden');
+            this.newFeatureTabLink.click();
+
+            if (!this.newFeatureWizard) {
```
Comment (steveoh) [inline src/app/project/FeatureDetails.js]: this needs to check for if the wizard is `_destroyed`

Comment (stdavis) [inline src/app/project/FeatureDetails.js]: Fixed with: f9b966d

**Comment by steveoh** [comment]
:sparkles: :tada: :shipit: :tada: :sparkles:

**Pull request merged**